docs/index.rst

```diff
@@ -129,54 +130,132 @@ Card Layout
 Each panel contains a card, which can itself contain a header and/or footer,
 split by three or more `=` and `.` respectively.
 
-.. tip::
+.. seealso::
 
-    For card colouring, it is advised to use the bootstrap contextual classes:
-    `bg-primary`, `bg-success`, `bg-info`, `bg-warning`, `bg-danger`, `bg-secondary`, `bg-dark` and `bg-light`.
+    The bootstrap 4 `card documentation <https://getbootstrap.com/docs/4.0/components/card/>`_,
+    and this `card tutorial <https://www.w3schools.com/bootstrap4/bootstrap_cards.asp>`_
 
 .. code-block:: rst
 
     .. panels::
-        :card: shadow bg-primary
 
         panel 1 header
         ==============
 
         panel 1 content
 
-        ...
+        more content
+
+        ..............
         panel 1 footer
 
         ---
-        column += text-center
-        card = bg-info
-        title = bg-success
-        footer = bg-secondary
 
+        panel 2 header
+        ==============
+
+        panel 2 content
+
+        ..............
+        panel 2 footer
+
+.. panels::
+
+    panel 1 header
+    ==============
+
+    panel 1 content
+
+    more content
+
+    ..............
+    panel 1 footer
+
+    ---
+
+    panel 2 header
+    ==============
+
+    panel 2 content
+
+    ..............
+    panel 2 footer
+
+
+Card Styling
+------------
+
+To style the look of cards,
+you may use the directive options to add default CSS classes for eac element,
+or use the per-panel option syntax to add to or override these:
+
+- container: the top-level container
+- column: the panel container
+- card: the panel card
+- body: the panel card
+- header: the panel header
+- footer: the panel footer
+
+You can add your own CSS (see
+`the html_css_files option <https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-html_css_files>`_)
+but it is advised you use the built-in bootstrap classes:
+
+- `Card colouring <https://getbootstrap.com/docs/4.0/utilities/colors/>`_  contextual classes: `bg-primary`, `bg-success`, `bg-info`, `bg-warning`, `bg-danger`, `bg-secondary`, `bg-dark` and `bg-light`.
+- `Padding and margins <https://getbootstrap.com/docs/4.0/utilities/spacing/>`_: `border-0`, `p-2`, `m-2`, ...
```

Comment:
This makes me wonder if there'd be value in having something like "sphinx-bootstrap" that is just an easy way to define a few useful bootstrap elements with sphinx directives. You could imagine instructions like these being applied to a number of UI elements. (as opposed to having one sphinx extension per element type, such as "sphinx-panels", "sphinx-togglebutton", etc)

Reply:
yep, if this existed it would be easy to add the dependency to this package. If you look at the CSS I have added, you'll see that I have divided into approximately the classes required for each aspect.

Reply:
so there'd be something like a bootstrap meta-package that would make it possible for extension to depend on it and then those extensions don't have to worry about loading any bootstrap JS/CSS themselves?

That might be nice, then we could look into re-working `sphinx-togglebutton` so that it uses bootstrap, but without requiring users to use a theme that has bootstrap

Reply:
Another thing that could benefit from bootstrap would be "question and answer" blocks: https://getbootstrap.com/docs/4.1/components/forms/